test/Microsoft.Azure.WebJobs.Host.EndToEndTests/AsyncCancellationEndToEndTests.cs

```diff
@@ -22,13 +22,17 @@ public class AsyncCancellationEndToEndTests : IDisposable
         private static EventWaitHandle _functionStarted;
         private static EventWaitHandle _functionCompleted;
         private static bool _tokenCancelled;
+        private static SynchronizationContext _oldContext;
 
         private readonly CloudStorageAccount _storageAccount;
         private readonly RandomNameResolver _resolver;
         private readonly JobHostConfiguration _hostConfiguration;
 
         public AsyncCancellationEndToEndTests()
         {
+            // Run tests in multithreaded environment
+            _oldContext = SynchronizationContext.Current;
+            SynchronizationContext.SetSynchronizationContext(null);
```

Comment:
Sorry, I misread the diff here. But after chatting, I do think it would be a better approach and (safer in the long run) if we were to scope this per method, ensuring the context is restored at the end of the method call.